ga4gh/frontend.py

```diff
@@ -267,6 +307,43 @@ def assertCorrectVersion(version):
         raise exceptions.VersionNotSupportedException()
 
 
+def startLogin():
+    flask.session["state"] = oic.oauth2.rndstr(SECRET_KEY_LENGTH)
+    flask.session["nonce"] = oic.oauth2.rndstr(SECRET_KEY_LENGTH)
+    args = {
+        "client_id": app.oidcClient.client_id,
+        "response_type": "code",
+        "scope": ["openid", "profile"],
+        "nonce": flask.session["nonce"],
+        "redirect_uri": app.oidcClient.redirect_uris[0],
+        "state": flask.session["state"]
+    }
+
+    result = app.oidcClient.do_authorization_request(
+        request_args=args, state=flask.session["state"])
+    return flask.redirect(result.url)
+
+
+@app.before_request
+def checkAuthentication():
+    """
+    The request should come with an Authorization header of form:
+    Authorization: Bearer <token> if it's from the command line, or
+    have a session key of 'token' if it's the browser.
+    If the token is not found, start the login process.
+
+    If there is no oidcClient, we are running naked and we don't check.
+    If we're being redirected to the oidcCallback we don't check.
+    """
+    if app.oidcClient is not None \
```

Comment:
I'm confused by this function: what does it return? Does it raise an exception if the request isn't authenticated? If we are returning None as the success indicator, we should be clearer about this and use this to be more explicit about the structure using early returns. e.g.
```python
# Authentication is not enabled, so all requests are authorized.
if app.oidcClient is None:
    return None
# The request was sent to the oidcCallback, and is therefore authorised by definition.
if flask.request.endpoint == 'oidcCallback':
    return None
# etc
```